presto-main/src/main/java/com/facebook/presto/sql/planner/PlanNodeCanonicalInfo.java

```diff
@@ -84,8 +84,8 @@ public static List<CanonicalPlanWithInfo> getCanonicalInfo(
                     continue;
                 }
                 PlanNode statsEquivalentPlanNode = node.getStatsEquivalentPlanNode().get();
-                Optional<String> hash = planCanonicalInfoProvider.hash(session, statsEquivalentPlanNode, strategy);
-                Optional<List<PlanStatistics>> inputTableStatistics = planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode);
+                Optional<String> hash = planCanonicalInfoProvider.hash(session, statsEquivalentPlanNode, strategy, true);
+                Optional<List<PlanStatistics>> inputTableStatistics = planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode, true);
```

Comment:
This function is called at the end to collect the HBO information, hence `cacheOnly` set to true.